src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> item = new Node<>(tail, value, null);
+        if (head == null) {
+            head = item;
+        } else {
+            tail.next = item;
+        }
+        tail = item;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> indexNode = getNode(index);
+            if (indexNode == head) {
+                Node<T> newNode = new Node<>(null, value, indexNode);
+                indexNode.prev = newNode;
+                head = newNode;
+            } else {
+                Node<T> newNode = new Node<>(indexNode.prev, value, indexNode);
+                indexNode.prev.next = newNode;
+                indexNode.prev = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> indexNode = getNode(index);
+        T oldValue = indexNode.item;
+        indexNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> indexNode = getNode(index);
+        if (head == indexNode && tail == indexNode) {
+            head = null;
+            tail = null;
+        } else if (head == indexNode) {
+            indexNode.next.prev = null;
+            head = indexNode.next;
+        } else if (tail == indexNode) {
```

Comment:
too much else if. try to think about two if-else constructions: if-else if-else, without `else if`